### PR TITLE
Change WorkflowData from interface to class

### LINKED ISSUE #53: [FEATURE] Simplify WorkflowData to a concrete class
### Is your feature request related to a problem?

`WorkflowData` was created as an interface, originally envisioning multiple types of data.  However, during development we realized it was far easier to just pass around the same single class.  Because it's an interface, this requires anonymous subclassing.

Creating an anonymous subclass returning values generated at runtime can be dangerous.  One can spend several hours trying to debug why the following sequence hangs the JVM:

```java
final String s = inputIndex;
WorkflowData wfd = new WorkflowData() {
    @Override
    public Map<String, Object> getContent() {
        return Map.of("index", s);
    }
};
wfd.getContent(); // hangs creating Map.of if s was null
```
Using `Map.of()` with a null key or value causes a `NullPointerException` which stopped processing without any information on why.

### What solution would you like?

Just create a regular class and pass the params and content in as arguments to the constructors.   We can even overload the constructors to allow both, or just params, to default to an empty map.

### What alternatives have you considered?

Some sort of wrapper class around actual `Request` and `Response` objects in OpenSearch.  This is overly complicated for our purpose.

### Do you have any additional context?

I'd add this to my #47 but the changes also need to be made in #38, #44, and #52.  I suggest the PRs that are ready to merge get merged, we fix this, and future development can conform to the new signature before we do any more.

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Simplifies WorkflowData by making it a POJO.

Note: Impacts PRs https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/47, https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/38, https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/44, and https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/52.  Happy to rebase this after those are merged if they are ready to go, or we can merge this and rebase those.

### Issues Resolved

Fixes #53 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dbwiddis** [comment]
Note the changes required in the other PRs are minimal/trivial, so merging this first if there's still any iteration on your PRs will be a benefit.
```diff
- future.complete(new WorkflowData() {
-     @Override
-     public Map<String, Object> getContent() {
-         return Map.of("index-name", createIndexResponse.index());
-     }
- });
+ future.complete(new WorkflowData(Map.of("index-name", createIndexResponse.index())));
```

**Review by joshpalis — APPROVED** [review]

**Comment by owaiskazi19** [comment]
> Note the changes required in the other PRs are minimal/trivial, so merging this first if there's still any iteration on your PRs will be a benefit.
> 
> ```diff
> - future.complete(new WorkflowData() {
> -     @Override
> -     public Map<String, Object> getContent() {
> -         return Map.of("index-name", createIndexResponse.index());
> -     }
> - });
> + future.complete(new WorkflowData(Map.of("index-name", createIndexResponse.index())));
> ```

Ah! Missed this. Let me take care of this in my followup PR.

File: src/main/java/org/opensearch/flowframework/template/TemplateParser.java [hunk 1]
```diff
@@ -76,21 +76,10 @@ public static List<ProcessNode> parseJsonGraphToSequence(String json, Map<String
...
-                    public Map<String, String> getParams() {
-                        // See RestCreateIndexAction for source of param keys needed
-                        return Map.of("index", request.index());
-                    }
-
-                };
+                inputData = new WorkflowData(
+                    Map.of("mappings", request.mappings(), "settings", request.settings(), "aliases", request.aliases()),
+                    Map.of("index", request.index())
+                );
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply src/main/java/org/opensearch/flowframework/template/TemplateParser.java:82]: Yes, I'll iterate on that in #47.

This is unrelated to this PR, just enough to make it compile.

[the remark this answers is not in the document]
Comment (owaiskazi19) [reply src/main/java/org/opensearch/flowframework/template/TemplateParser.java:82]: Agreed! This is unrelated to this PR. Just a thought :)

[the remark this answers is not in the document]
Comment (dbwiddis) [reply src/main/java/org/opensearch/flowframework/template/TemplateParser.java:82]: Actually the whole `TemplateParser` class is deleted in #47.... .again, this is just to get it to compile but also shows the simplicity of the change.

**Review by owaiskazi19 — APPROVED** [review]
Thanks for this change!

**Comment by joshpalis** [comment]
@owaiskazi19 I can handle the CreateIndexStep modifications as part of #44

**Pull request merged**